### ISSUE TITLE
validate_call on method + inheritance bug

### ISSUE THREAD

**Issue by leandro-twigeo** [pydantic/pydantic#8146]
### Initial Checks

- [x] I confirm that I'm using Pydantic V2

### Description

A weird interaction breaks inheritance when using ``@validate_call`` with a method that is implemented on both a ``Parent`` class and ``Child`` class that inherits from ``Parent``, with the ``Child`` class method calling the ``Parent`` method via  ``super()``.

As per my code example, the first time a ``Child`` object calls the method ``do_something``, everything works fine, but the second time, the ``Child.do_something`` method has been overwritten by the ``Parent.do_something``method. I've tried few combinations:
* If I remove ``@validate_call`` and keep ``super().do_something`` in the child method, things work properly (pure python, works expected);
* If I keep ``@validate_call`` and remove ``super().do_something`` in the child method, things work properly;
* If I keep ``@validate_call`` and replace ``super().do_something`` by ``Parent.do_something(self, ...)`` in the child method, things work properly;

I believe that the ``super().do_something(a=a, b=b)`` call within the ``Child.do_something`` method triggers some validation that binds the ``do_something`` method of the ``Parrent`` class to the ``Child`` class.

### Example Code

```Python
import inspect

from pydantic import validate_call


class Parent:
    @validate_call
    def do_something(self, a: int, b: int):
        print(f"PARENT: {a=}, {b=}")


class Child(Parent):
    @validate_call
    def do_something(self, a: int, b: int, c: int):
        super().do_something(a=a, b=b)
        print(f"CHILD: {c=}")


print("Testing the Parent class:")
parent_obj = Parent()
for i in range(3):
    parent_obj.do_something(a=3, b=7)


print("\nTesting the Child class:")
child_obj = Child()
for i in range(3):
    try:
        print(inspect.getfullargspec(child_obj.do_something.raw_function))
    except:
        pass
    child_obj.do_something(a=3, b=7, c=10)
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.5.1
        pydantic-core version: 2.14.3
          pydantic-core build: profile=release pgo=true
                 install path: /Users/leandro/miniconda3/envs/twigeox-dev/lib/python3.10/site-packages/pydantic
               python version: 3.10.13 (main, Sep 11 2023, 08:16:02) [Clang 14.0.6 ]
                     platform: macOS-14.0-arm64-arm-64bit
             related packages: typing_extensions-4.8.0
```

**Comment by waterfountain1996** [comment]
Looks like it works without the following lines:
https://github.com/pydantic/pydantic/blob/f42dc2fe978d88c9d06796e0a1837fe5d0b13507/pydantic/_internal/_validate_call.py#L135-L139

The tests still pass locally, not sure what impact it has on perfomance

**Comment by sydney-runkle** [comment]
Hi @leandro-twigeo,

Thanks for reporting this. Yep, looks like a bug introduced in `2.5` when we fixed the lack of support for `validate_call` on methods in classes with `__slots__` defined. I'll look into a fix for this bug today 🐛!

**Comment by leandro-twigeo** [comment]
Thanks for looking at this @sydney-runkle! 

To add extra info, I actually had Pydantic v2.4.X when getting the problem the first time. I then upgraded to see if it persisted (and it did). I've just tried to run my code example with older versions than 2.5, and as far as I could see, I had the same bug with pydantic-2.4.2, pydantic-2.3.0 and pydantic-2.2.1. Just letting you know in case it helps you :)
